File: py/hello.py
#комментарий
print("Hello!", "How are you?"); print("Bye!")
print("Result: %o." % (10 + 5))
print("Hex->Dec:", 0x10)
print("Oct->Dec:", 0o10)
print("Bin->Dec:", 0b10)
#                  ^
#           двоичный литерал
print("Hex->Oct: %o" % 0x10)  
print("Hex->Oct:", oct(0x10))
print("Hex->Bin:", bin(0x10))

a = 0x10    #переменная
print(a)
a = 0b10
print(a)
print(type(a))
a = "aaaaaaa"
print(a, type(a))

for i in range(5):
    print(i)

print()

for i in range(-4, 5, 2):
    print(i)
    print("|")

print()

N = 100
s = 0
for i in range(1, N + 1):
    s += i
#    s = s + i
print(s)

print()

print(5 / 3)
print(5 // 3)
print(-5 // 3)
print(5 % 3)
print(-5 % 3)

print(bin(87626798))

File: cpp/hello.c
//комментарий
#include <stdio.h>

int main() {
    printf("Hello! How are you?\n"); //комментарий
    printf("Result: %o, %x.\n", 10 + 5, 500 - 33);
//         <-               ->
//          форматная строка
    printf("String: %s.\n", "aaaaaa"); 
//                   ^
//              спецификатор
    printf("Hex->Oct: %o\n", 0x10);
    printf("Hex->Dec: %d\n", 0x10);
    printf("Oct->Dec: %d\n", 010);
//                           ^
//                 восьмеричный литерал
    
    int a;      //definition (определение)
//      ^
// идентификатор переменной
    a = 0x10;
    int b = 011;   //инициализация
    int d = 111, e = 222, f;
    printf("a: %d\n", a);
    printf("f: %d\n", f);
    a = 010;
    printf("a: %d\n", a);
    a = 3000000000;
    printf("a: %d\n", a);
    printf("sizeof a: %ld\n", sizeof a);
    long c = 3000000000;
    printf("c: %ld\n", c);
    printf("sizeof c: %ld\n", sizeof c);

    for(int i = 0; i < 5; i++) {
        printf("i: %d\n", i);
    }



    printf("\n");

    int N = 100;
    int s = 0;
    for(int i = 1; i <= N; i++) {
        s += i;
    }
    printf("%d\n", s);

    printf("\n");
    printf("%d\n", 5 / 3);
    printf("%d\n", -5 / 3);
    printf("%d\n", 5 % 3);
    printf("%d\n", -5 % 3);
    printf("%d\n", -9 % 8);
}
